zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala

```diff
@@ -198,11 +198,18 @@ final case class Endpoint[PathInput, Input, Err, Output, Auth <: AuthType](
   def auth[Auth0 <: AuthType](auth: Auth0): Endpoint[PathInput, Input, Err, Output, Auth0] =
     copy(authType = auth)
 
-  def authScopes: List[String] = authType match {
-    case AuthType.ScopedAuth(_, scopes) => scopes
-    case _                              => Nil
+  def authScopes(authType: AuthType): List[String] = authType match {
+    case AuthType.ScopedAuth(nestedAuth, scopes) =>
+      scopes ++ authScopes(nestedAuth)
+    case AuthType.Or(auth1, auth2, _)            =>
+      authScopes(auth1) ++ authScopes(auth2)
+    case _                                       =>
+      Nil
   }
 
+  def scopes(scopes: String*): Endpoint[PathInput, Input, Err, Output, AuthType] =
```

Comment:
This method should be called with at least one scope and only if the authType is not `AuthType.None`